lib/IRGen/IRGenModule.cpp

```diff
@@ -1104,7 +1104,7 @@ void IRGenModule::registerRuntimeEffect(ArrayRef<RuntimeEffect> effect,
 #define NO_ATTRS {}
 #define EFFECT(...) { __VA_ARGS__ }
 #define NO_MEMEFFECTS                                                          \
-  { llvm::MemoryEffects::none() }
+  { llvm::MemoryEffects::unknown() }
```

Comment:
I defer to @aschwaighofer on how he wants to handle this.

Reply:
Yeah, me too :). But as far as I understand it, this matches the behavior we had. `unknown` is the equivalent to not specifying where as `none` is `ReadNone` (you can see this in the test updates in https://reviews.llvm.org/D135780).

To put another way:
  * `ReadNone` -> `MemoryEffects::none()`
  * `ReadOnly` -> `MemoryEffects::readOnly()`
  * `WriteOnly` -> `MemoryEffects::writeOnly()`
  * None specified -> `MemoryEffects::unknown()`

Reply:
Yep that looks good.

```c++
    // Upgrade old memory attributes.
    MemoryEffects ME = MemoryEffects::unknown();
    if (Attrs & (1ULL << 9)) {
      // ReadNone
      Attrs &= ~(1ULL << 9);
      ME &= MemoryEffects::none();
    }
    if (Attrs & (1ULL << 10)) {
      // ReadOnly
      Attrs &= ~(1ULL << 10);
      ME &= MemoryEffects::readOnly();
    }
```